pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1301,7 +1302,9 @@ func buildListener(opts buildListenerOpts, trafficDirection core.TrafficDirectio
 
 		if opts.proxy.Type != model.Router {
 			res.ListenerFiltersTimeout = opts.push.Mesh.ProtocolDetectionTimeout
-			if res.ListenerFiltersTimeout != nil {
+			// if timeout is not disabled, set ContinueOnListenerFiltersTimeout
+			// if timeout = nil, a default 15s will be applied.
+			if res.ListenerFiltersTimeout == nil || !(res.ListenerFiltersTimeout.GetNanos() == 0 && res.ListenerFiltersTimeout.GetSeconds() == 0) {
```

Comment:
I know it was there earlier but do we have to check this at all? Why do not we set `ContinueOnListenerFiltersTimeout` by default?

Reply:
Do you mean always make ContinueOnListenerFiltersTimeout = true even no timeout is set? From the doc, there is no harmful effect said

Reply:
Yes. I think that simplifies this and Envoy will use that flag only when listener filter times out.